keyboards/ymdk/ymd09/keymaps/patrickgilsf/keymap.c

```diff
@@ -0,0 +1,431 @@
+#include QMK_KEYBOARD_H
+//#include "raw_hid.h"
+
+
+
+enum custom_keycodes {
+    KEY1 = SAFE_RANGE,
+    KEY2,
+    KEY3,
+    KEY4,
+    KEY5,
+    KEY6,
+    KEY7,
+    KEY8,
+    KEY9,
+    KEY10,
+    KEY11,
+    KEY12,
+    KEY13,
+    KEY14,
+    KEY15,
+    KEY16,
+    KEY17,
+    KEY18,
+    KEY19,
+    KEY20,
+    KEY21,
+    KEY22,
+    KEY23,
+    KEY24,
+    KEY25,
+    KEY26,
+    KEY27
+};
+
+
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+  [0] = LAYOUT(KEY1, KEY2, KEY3, 
+               KEY4, KEY5, KEY6, 
+               KEY7, KEY8, KEY9),
+
+  [1] = LAYOUT(KEY10, KEY11, KEY12,
+               KEY13, KEY14, KEY15,
+               KEY16, KEY17, KEY18),
+
+  [2] = LAYOUT(KEY19, KEY20, KEY21,
+               KEY22, KEY23, KEY24,
+               KEY25, KEY26, KEY27)
+
+};
+
+//this is the timer for flash mode sequence
+//static uint32_t key_timer = 0;
+
+
+
+
+/*
+Legend:
+--Layer 0 - [Mac OS Zoom shortcuts]
+KEY1 - Mute Mic/Toggle Keyboard Light;
+KEY2 - Mute Video;
+KEY3 - Raise/Lower Hand;
+KEY4 - Speaker/Gallery View Toggle
+KEY5 - Prev Page of Gallery
+KEY6 - Next Page of Gallery
+KEY7 - Full Screen
+KEY8 - Show/Hide Chat; // If KEY7 is down, up to layer 1.
+KEY9 - Minimal Window;
+
+--Layer 1 - 
+KEY10 - Mute Mic/Toggle Keyboard Light;
+KEY11 - Mute Video; // If KEY10 is down, back to layer 0
+KEY12 - Raise/Lower Hand;
+KEY13 - Speaker/Gallery View Toggle
+KEY14 - Prev Page of Gallery
+KEY15 - Next Page of Gallery
+KEY16 - Full Screen
+KEY17 - Show/Hide Chat; // If KEY7 is down, down to layer 0.
+KEY18 - Minimal Window; // If KEY7 is down, up to layer 2
+
+-- Layer 2 -
+KEY19 - 
+KEY20 - 
+KEY21 - 
+KEY22 -
+KEY23 -
+KEY24 -
+KEY25 - 
+KEY26 - If KEY25 is down, back to layer 1
+KEY27 - If KEY25 && KEY26 are down, puts it in setup mode
+*/
+
+//initialize booleans for stacked functions:
+bool kp_1 = false;
```

Comment:
you could use `bool key[26] = {0};` for this instead.

Reply:
Thanks for the advice. I'm not that familiar with C, mostly JS and node.

Reply:
@drashna are my changes ok in the new push? Will this FW get merged at any point? TY.